api/events/schema.py

```diff
@@ -218,15 +219,38 @@ class Mutation(graphene.ObjectType):
              event[required]")
 
 
+class PaginatedDailyRoomEvents(graphene.ObjectType):
+    """
+    Paginated result for daily room events
+    """
+    DailyRoomEvents = graphene.List(DailyRoomEvents)
+    has_previous = graphene.Boolean()
+    has_next = graphene.Boolean()
+    pages = graphene.Int()
+    query_total = graphene.Int()
+
+
 class Query(graphene.ObjectType):
-    all_events = graphene.List(
-        DailyRoomEvents,
+    all_events = graphene.Field(
+        PaginatedDailyRoomEvents,
         start_date=graphene.String(),
         end_date=graphene.String(),
-        description="Query that returns daily room events")
+        page=graphene.Int(),
+        per_page=graphene.Int(),
+        description="Query that returns paginated daily room events")
 
     @Auth.user_roles('Admin', 'Default User')
     def resolve_all_events(self, info, **kwargs):
+        page = kwargs.get('page')
+        per_page = kwargs.get('per_page')
+        if page is not None and page < 1:
```

Comment:
Just us `if page   and page< 1:`

Reply:
@Bonifase Using `if page and page < 1` will not catch zeros.